test/enums.jl

```diff
@@ -179,6 +179,11 @@ end
 @enum HashEnum2 Enum2_a=1
 @test hash(Enum1_a) != hash(Enum2_a)
 
+# PR #49777: Check that `Base.hash` can be specialized by the user without
+# overwriting a method definition.
+@enum HashEnum3 Enum3_a=1
+@test only(methods(hash, (HashEnum3, UInt))).sig != Tuple{typeof(hash), HashEnum3, UInt64}
```

Comment:
@stev47: I was thinking of a test like this. Can we include both the one above, and this one?:
```suggestion
@enum HashEnum3 Enum3_a=1
@test only(methods(hash, (HashEnum3, UInt))).sig != Tuple{typeof(hash), HashEnum3, UInt64}

# Now test overriding the hash with a custom hash and ensure that it's dispatched to.
Base.hash(::HashEnum3, UInt) = 0
@test hash(ENUM3_a) === 0
```

It's kind of a silly test, as you note, but it seems prudent to have the expected user flow in a test.

Reply:
Since it will succeed either way, it does not seem to test anything useful though? Basic method definition is being tested somewhere else already I believe.

But I don't mind adding it if @KristofferC thinks that it is good to have.

Reply:
`only(methods())` should be `which()` instead

Reply:
> `only(methods())` should be `which()` instead

thanks, fixed

Reply:
Yeah, i'm convinced. The test you have is sufficient, assuming that method definition works as expected. 👍 

K, sgtm. I'll resolve this thread.